examples/step-37/step-37.cc

```diff
@@ -1094,6 +1094,7 @@ namespace Step37
 
     time.reset();
     time.start();
+    constraints.set_zero(solution);
```

Comment:
Can you explain this change? In this case, the solution is subject to zero Dirichlet boundary conditions, and the solution vector is zero by construction, so this is not necessary. Furthermore, the preconditioner should scale the constrained entries in a way that the preconditioned operator contains ones on the diagonal (at least I hope so) and thus right in the correct magnitude of the spectrum.

Reply:
In text it’s stated that code is ready for hanging nodes and can be extended to non-zero bc. 

At least for both together this is needed

Reply:
p.s. i just thought that it's better to have this because if someone tries to copy-paste extension possibilities, the solve won't work (at least that's my experience, see the .cc file I sent you privately with regards to that other issue we discuss).

Reply:
@kronbichler ping.